mono/metadata/threads.c

```diff
@@ -5065,15 +5088,67 @@ threads_add_joinable_thread_nolock (gpointer tid)
 }
 #endif
 
+static void
+threads_add_pending_joinable_runtime_thread (gpointer thread_info)
+{
+	g_assert (thread_info);
+	MonoThreadInfo *mono_thread_info = (MonoThreadInfo*)thread_info;
```

Comment:
Nitpick, how about moving the cast to the caller side?

Reply:
Absolutely, this method was initially included in the header file with mono_threads_add_joinable_runtime_thread that is using gpointer due to some include problems. When I moved it into the source file converting it to static, the signature was kept. Will change.